client/src/components/SpendingForm.js

```diff
@@ -50,151 +44,134 @@ const Title = styled.p`
 const Container1 = styled.div`
   display: flex;
   justify-content: space-between;
-`;
-
-const Container2 = styled.div`
-  display: flex;
-  justify-content: center;
-  width: 100%;
-`;
-
-const Container3 = styled.div`
-  display: flex;
-  flex-direction: column;
-  align-items: center;
-  margin-bottom: 40px;
-`;
-
-const Headings = styled.p`
-  all: unset;
-  margin-top: 10px;
-  font-size: 14px;
-  margin-right: auto;
-  margin-left: 5px;
-`;
-
-const CategorysContainer = styled.div`
-  display: flex;
-  align-items: space-between;
-  margin-top: 5px;
   width: 100%;
 `;
 
-const CategoryTitle = styled.p`
-  all: unset;
-  font-size: 7pt;
-  color: ${light.colors.fontsecondary};
-  text-align: center;
-  margin-top: 10px;
-`;
-
 const CategoryContainer = styled.div`
   display: flex;
-  flex-direction: column;
+  flex-wrap: wrap;
+  justify-content: space-between;
   width: 100%;
-  margin-left: 0;
-  margin-bottom: 10px;
+  margin-top: -20px;
 `;
 
-const styleFirstInput = {
-  marginLeft: 0
+const styleFormItem = {
+  flex: '0 0 48%'
 };
 
-const styleSecondInput = {
-  marginRight: 0
-};
+const categories = [
+  {
+    icon: 'food',
+    value: 'food',
+    fill: light.colors.action
+  },
+  {
+    icon: 'transport',
+    value: 'transport',
+    fill: light.colors.action
+  },
+  {
+    icon: 'airplane',
+    value: 'airplane',
+    fill: light.colors.action
+  },
+  {
+    icon: 'hotel',
+    value: 'hotel',
+    fill: light.colors.action
+  },
+  {
+    icon: 'drinks',
+    value: 'drinks',
+    fill: light.colors.action
+  },
+  {
+    icon: 'shopping',
+    value: 'shopping',
+    fill: light.colors.action
+  },
+  {
+    icon: 'action',
+    value: 'action',
+    fill: light.colors.action
+  },
+  {
+    icon: 'smiley',
+    value: 'smiley',
+    fill: light.colors.action
+  }
+];
 
-const styleButtonRight = {
-  marginRight: '5px'
-};
+export default function SpendingForm({ handleClick, handleClose, value }) {
+  const [title, setTitle] = useState('');
+  const [amount, setAmount] = useState(null);
+  const [date, setDate] = useState('');
+  const [location, setLocation] = useState('');
+  const [category, setCategory] = useState('');
```

Comment:
Maybe the code gets a bit smaller if you set the Dates from the inputfields with one state.